contracts/config/ConfigSyncer.sol

```diff
@@ -0,0 +1,173 @@
+                updateApplied = true;
+
+                if (updateId > latestUpdateId) {
+                    latestUpdateId = updateId;
+                }
+            }
+
+            EventUtils.EventLogData memory eventData;
+
+            eventData.uintItems.initItems(3);
+            eventData.uintItems.setItem(0, "updateId", updateId);
+            eventData.uintItems.setItem(1, "prevValue", prevValue);
+            eventData.uintItems.setItem(2, "nextValue", updatedValue);
+
+            eventData.boolItems.initItems(1);
+            eventData.boolItems.setItem(0, "updateApplied", updateApplied);
+            
+            eventEmitter.emitEventLog(
+                "SyncConfig",
+                eventData
+            );
+        }
+        
+        if (latestUpdateId > dataStore.getUint(Keys.syncConfigLatestUpdateIdKey())) {
+            dataStore.setUint(Keys.syncConfigLatestUpdateIdKey(), latestUpdateId);
+        }
+    }
+
+    // @dev initialize the allowed base keys
+    function _initAllowedBaseKeys() internal {
+        allowedBaseKeys[Keys.MAX_POOL_AMOUNT] = true;
+        allowedBaseKeys[Keys.MAX_POOL_USD_FOR_DEPOSIT] = true;
+        allowedBaseKeys[Keys.MAX_OPEN_INTEREST] = true;
+
+        allowedBaseKeys[Keys.POSITION_IMPACT_FACTOR] = true;
+        allowedBaseKeys[Keys.POSITION_IMPACT_EXPONENT_FACTOR] = true;
+
+        allowedBaseKeys[Keys.SWAP_IMPACT_FACTOR] = true;
+        allowedBaseKeys[Keys.SWAP_IMPACT_EXPONENT_FACTOR] = true;
+
+        allowedBaseKeys[Keys.FUNDING_INCREASE_FACTOR_PER_SECOND] = true;
+        allowedBaseKeys[Keys.FUNDING_DECREASE_FACTOR_PER_SECOND] = true;
+        allowedBaseKeys[Keys.MIN_FUNDING_FACTOR_PER_SECOND] = true;
+        allowedBaseKeys[Keys.MAX_FUNDING_FACTOR_PER_SECOND] = true;
+
+        allowedBaseKeys[Keys.OPTIMAL_USAGE_FACTOR] = true;
+        allowedBaseKeys[Keys.BASE_BORROWING_FACTOR] = true;
+        allowedBaseKeys[Keys.ABOVE_OPTIMAL_USAGE_BORROWING_FACTOR] = true;
+        allowedBaseKeys[Keys.BORROWING_FACTOR] = true;
+        allowedBaseKeys[Keys.BORROWING_EXPONENT_FACTOR] = true;
+
+        allowedBaseKeys[Keys.RESERVE_FACTOR] = true;
+        allowedBaseKeys[Keys.OPEN_INTEREST_RESERVE_FACTOR] = true;
+    }
+
+    // @dev validate that the baseKey is allowed to be used
+    // @param baseKey the base key to validate
+    function _validateKey(bytes32 baseKey) internal view {
+        if (!allowedBaseKeys[baseKey]) {
+            revert Errors.InvalidBaseKey(baseKey);
+        }
+    }
+
+    // @dev validate that the market within data is equal to market
+    // @param baseKey the base key to validate
+    // @param market the market address
+    // @param data the data used to compute fullKey
+    function _validateMarketInData(bytes32 baseKey, address market, bytes memory data) internal pure {
+        address marketFromData;
+        if (baseKey == Keys.MAX_PNL_FACTOR) {
+            bytes32 unusedBytes32;
+            bool unusedBool;
```

Comment:
i don't think these are needed, could just comment out the unused variables, example: https://github.com/gmx-io/gmx-synthetics/blob/main/contracts/token/TokenUtils.sol#L62

Reply:
I've updated per the example provided in the latest commit. I also changed the local variable naming from unusedBytes32 and unusedBool to extKey and isLong respectively, for more informative naming. 

Variable name 'extKey' stands for extension or extended key, which is the naming I created and used in utils/configSyncer.ts for the additional key used for parameters with baseKey = MAX_PNL_FACTOR as these were unnamed in the updateMarketConfigUtils.ts file I derived them from and I wanted to differentiate them from baseKey to avoid confusion.

If preferred, I can revert this naming change.

Reply:
I think this is resolved, but I will let you confirm on your end in case I am somehow missing something here.